crates/sui-core/src/authority.rs

```diff
@@ -2123,11 +2126,40 @@ impl AuthorityState {
 pub struct ConsensusHandler {
     state: Arc<AuthorityState>,
     sender: Sender<ConsensusListenerMessage>,
+    hash: Mutex<u64>,
 }
 
 impl ConsensusHandler {
     pub fn new(state: Arc<AuthorityState>, sender: Sender<ConsensusListenerMessage>) -> Self {
-        Self { state, sender }
+        let hash = Mutex::new(0);
+        Self {
+            state,
+            sender,
+            hash,
+        }
+    }
+
+    fn update_hash(&self, index: ExecutionIndices, v: &[u8]) -> ExecutionIndicesWithHash {
+        let mut hash_guard = self
+            .hash
+            .try_lock()
+            .expect("Should not have contention on ExecutionState::update_hash");
+        let mut hasher = DefaultHasher::new();
+        (*hash_guard).hash(&mut hasher);
+        v.hash(&mut hasher);
+        let hash = hasher.finish();
+        *hash_guard = hash;
+        // Log hash for every 100th sequenced certificate
+        if index.next_transaction_index == 1
+            && index.next_batch_index == 1
+            && index.next_certificate_index % 100 == 0
+        {
+            debug!(
```

Comment:
we already have a debug log in handle_consensus_transaction for every cert, I see no reason not to log every single cert for now.